schema/encoding/jsonschema_test.go

```diff
@@ -0,0 +1,300 @@
+		Regexp: `\s+$`,
+	}
+	b := new(bytes.Buffer)
+	assert.NoError(t, validatorToJSONSchema(b, validator))
+	_, err := isValidJSON(wrapWithJSONObject(b))
+	assert.NoError(t, err)
+	a := assert.New(t)
+	a.Contains(b.String(), string([]byte{'\\', '\\', 's', '+', '$'}))
+}
+
+func TestStringValidator(t *testing.T) {
+	validator := &schema.String{
+		MinLen: 3,
+		MaxLen: 23,
+	}
+	b := new(bytes.Buffer)
+	assert.NoError(t, validatorToJSONSchema(b, validator))
+	m, err := isValidJSON(wrapWithJSONObject(b))
+	assert.NoError(t, err)
+
+	a := assert.New(t)
+	strJSON := string(wrapWithJSONObject(b))
+	// check string for values
+	a.Contains(strJSON, "minLength")
+	a.Contains(strJSON, "maxLength")
+	a.Contains(strJSON, `"type": "string"`)
+
+	// check decoded JSON for values
+	a.Equal(m["type"], "string")
+	a.Equal(m["minLength"], float64(validator.MinLen))
+	a.Equal(m["maxLength"], float64(validator.MaxLen))
+
+}
+
+func TestEmptyStringValidator(t *testing.T) {
+	validator := &schema.String{}
+	b := new(bytes.Buffer)
+	assert.NoError(t, validatorToJSONSchema(b, validator))
+	m, err := isValidJSON(wrapWithJSONObject(b))
+	assert.NoError(t, err)
+
+	a := assert.New(t)
+	strJSON := string(wrapWithJSONObject(b))
+	// check string for absence values
+	a.NotContains(strJSON, "minLength")
+	a.NotContains(strJSON, "maxLength")
+	// check string for values
+	a.Contains(strJSON, `"type": "string"`)
+
+	// check decoded JSON for values
+	a.Equal(m["type"], "string")
+	_, ok := m["minLength"]
+	a.Equal(ok, false)
+	_, ok = m["maxLength"]
+	a.Equal(ok, false)
+}
+
+func TestAllowedStringValidation(t *testing.T) {
+	validator := &schema.String{
+		Allowed: []string{"one", "two"},
+	}
+	b := new(bytes.Buffer)
+	assert.NoError(t, validatorToJSONSchema(b, validator))
+	m, err := isValidJSON(wrapWithJSONObject(b))
+	assert.NoError(t, err)
+
+	a := assert.New(t)
+	strJSON := string(wrapWithJSONObject(b))
+	a.NotContains(strJSON, "multipleOf")
+	a.Contains(strJSON, `"type": "string"`)
+	a.Contains(strJSON, `"enum"`)
+
+	a.Equal(m["type"], "string")
+	assert.Len(t, m["enum"], 2)
+	a.Equal(m["enum"], copyStringToInterface([]string{"one", "two"}))
+}
+
+func TestIntegerValidatorNoBoundaryPanic(t *testing.T) {
+	validator := &schema.Integer{}
+	// Catch regressions in Integer boundary handling
+	assert.NotPanics(t, func() { validatorToJSONSchema(new(bytes.Buffer), validator) })
+}
+
+func TestStringValidatorNoBoundaryPanic(t *testing.T) {
+	validator := &schema.String{}
+	// Catch regressions in Integer boundary handling
+	assert.NotPanics(t, func() { validatorToJSONSchema(new(bytes.Buffer), validator) })
+}
+
+func TestAllowedIntegerValidator(t *testing.T) {
+	validator := &schema.Integer{
+		Allowed: []int{10, 50},
+	}
+	b := new(bytes.Buffer)
+	assert.NoError(t, validatorToJSONSchema(b, validator))
+	m, err := isValidJSON(wrapWithJSONObject(b))
+	assert.NoError(t, err)
+
+	a := assert.New(t)
+	strJSON := string(wrapWithJSONObject(b))
+	a.Contains(strJSON, `"type": "number"`)
```

Comment:
this is float, I think it's ok. You were probably referring to an early patch set?